providers/common/ciphers/aes.c

```diff
@@ -0,0 +1,402 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h>
+#include <openssl/crypto.h>
+#include <openssl/core_numbers.h>
+#include <openssl/core_names.h>
+#include <openssl/evp.h>
+#include <openssl/params.h>
+#include "internal/cryptlib.h"
+#include "internal/provider_algs.h"
+#include "ciphers_locl.h"
+
```

Comment:
I would like to suggest that you add forward declarations like the ones I propose in 5bc017462982592f9fa0ba05f796c05418f5d05c.  This ensures that the functions are implemented with correct signatures.

Reply:
Done